go/mysql/replication/mysql56_gtid.go

```diff
@@ -29,6 +29,10 @@ import (
 // Mysql56FlavorID is the string identifier for the Mysql56 flavor.
 const Mysql56FlavorID = "MySQL56"
 
+var (
+	ErrExpectMysql56Flavor = vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "Expecting MySQL56 position flavor")
```

Comment:
I'm conflicted here, since this is user facing. I say that as "MySQL56" has no valid meaning for flavor or GTID format as we use it. 5.6 happens to be when MySQL added GTIDs and it's just "MySQL" vs say MariaDB. So for end user errors, I don't like using it. But then again... it's that specific string that we expect... 

In this case, perhaps this would be better:
```
ErrExpectMysql56Flavor = vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "expected MySQL GTID position but found invalid format. If the GTID position is for another database flavor then you must prefix the position string with that flavor, e.g. MariaDB/0-1-200")
```

Otherwise it's just perpetuating an internal mistake that we made (IMO) by calling the MySQL format anything other than MySQL. It's confusing for someone using MySQL 8.0 to specify that anything they're doing is MySQL56 related.

Reply:
Related to this: https://github.com/vitessio/vitess/pull/14977

I'm not sure it's worth doing though as it will be disruptive. But we can at least try to keep that nomenclature within Vitess as much as we can so that it's only relevant to developers of it rather than users of it.

Reply:
I took your suggestion but cut short the sentence to:

```go
	ErrExpectMysql56Flavor = vterrors.Errorf(vtrpc.Code_INVALID_ARGUMENT, "expected MySQL GTID position but found a different or invalid format.")
```

The reason is that we only accept MySQL GTID here, so I don't want to suggest to the user that they should prefix with a `MariaDB/`, as that will mislead them to think we will support it once prefixed.